core/src/execution/datafusion/expressions/cast.rs

```diff
@@ -644,14 +644,16 @@ impl Cast {
                     | DataType::Float32
                     | DataType::Float64
             ),
-            DataType::Decimal128(_, _) => matches!(
+            DataType::Decimal128(_, _) | DataType::Decimal256(_, _) => matches!(
```

Comment:
Hmm, I think Spark doesn't have decimal 256.

Reply:
If I remove the decimal 256 support here, I get this test failure:

```
- Fix NPE in partial decimal sum *** FAILED *** (119 milliseconds)
  org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 190.0 failed 1 times, most recent failure: Lost task 0.0 in stage 190.0 (TID 468) (192.168.86.33 executor driver): org.apache.comet.CometNativeException: Execution error: Comet Internal Error: Native cast invoked for unsupported cast from Decimal128(5, 2) to Decimal256(5, 2)
```

Reply:
I think this is probably the root cause (in `planner.rs`):

```rust
                // For some Decimal128 operations, we need wider internal digits.
                // Cast left and right to Decimal256 and cast the result back to Decimal128
                let left = Arc::new(Cast::new_without_timezone(
                    left,
                    DataType::Decimal256(p1, s1),
                    EvalMode::Legacy,
                ));
                let right = Arc::new(Cast::new_without_timezone(
                    right,
                    DataType::Decimal256(p2, s2),
                    EvalMode::Legacy,
                ));
```

Reply:
Ah, right, we temporarily upcast decimal 128 to decimal 256 to avoid overflow issue.